samples/devfiles/outerloop-dockerfile-build-inlined-deploy-devfile.yaml

```diff
@@ -0,0 +1,98 @@
+# Devfile based on the existing Node Sample https://github.com/nodeshift-starters/devfile-sample
+schemaVersion: 2.1.0
+metadata:
+  name: nodejs
+  version: 1.0.1
+  displayName: Node.js Runtime
+  description: Stack with Node.js 14
+  tags: ["NodeJS", "Express", "ubi8"]
+  projectType: "nodejs"
+  language: "nodejs"
+variables:
+  CONTAINER_IMAGE: node-image:latest
+  COMPONENT_NAME: my-node
+  PORT: "3001"
+components:
+  - name: outerloop-build
+    image:
+      imageName: "{{CONTAINER_IMAGE}}"
+      dockerfile:
+        uri: ./Dockerfile
+        buildContext: /project
+        rootRequired: false
+  - name: outerloop-deploy
+    kubernetes:
+      inlined: |
+        kind: Deployment
+        apiVersion: apps/v1
+        metadata:
+          name: {{COMPONENT_NAME}}
+        spec:
+          replicas: 1
+          selector:
+            matchLabels:
+              app: node-app
+          template:
+            metadata:
+              labels:
+                app: node-app
+            spec:
+              containers:
+                - name: {{COMPONENT_NAME}}
+                  image: {{CONTAINER_IMAGE}}
+                  ports:
+                    - name: http
+                      containerPort: {{PORT}}
+                      protocol: TCP
+                  resources:
+                    limits:
+                      memory: "128Mi"
+                      cpu: "500m"
+  - name: runtime
+    container:
+      image: registry.access.redhat.com/ubi8/nodejs-14:latest
+      memoryLimit: 1024Mi
+      mountSources: true
+      sourceMapping: /project
+      endpoints:
+        - name: http-3000
+          targetPort: 3000
+commands:
+  - id: install
+    exec:
+      component: runtime
+      commandLine: npm install
+      workingDir: /project
+      group:
+        kind: build
+        isDefault: true
+  - id: run
+    exec:
+      component: runtime
+      commandLine: npm start
+      workingDir: /project
+      group:
+        kind: run
+        isDefault: true
+  - id: debug
+    exec:
+      component: runtime
+      commandLine: npm run debug
+      workingDir: /project
+      group:
+        kind: debug
+        isDefault: true
+  - id: test
+    exec:
+      component: runtime
+      commandLine: npm test
+      workingDir: /project
+      group:
+        kind: test
+        isDefault: true
+  - id: deployk8s
+    apply:
+      component: outerloop-deploy
+      group:
+        kind: deploy
```

Comment:
The `deploy` command should be a `composite` command, referencing image build and k8s deploy.

Otherwise, based on the https://github.com/devfile/api/blob/main/docs/proposals/devfile/outer-loop-build-and-deploy.md the image will be build at the startup.
This is definitely not what most of the users will want to do. Because any changes done between startup and executing deploy command won't be reflected in the deployment. Users will be left wondering why they see something else in outer-loop, then what is running in inner-loop. 

All our examples and samples should have `kind: deploy` command that is `composite` command  chaining image build and deployment.  This is how big majority of  outer-loop flows will look like.  Or at least have `apply` command referencing `image` container, to make sure that the image is build when user requests it and not at the startup.  


```yaml 
  - id: deployk8s
    apply:
      component: outerloop-deploy
  - id: build-image
    apply:
      component: outerloop-build
  - id: deploy
    composite:
      commands:
        - build-image
        - deployk8s
      group:
        kind: deploy
```


There might be use case for building image at the startup, but I don't see it being that widely used.

Reply:
sounds good, I'll update it.

Reply:
updated!